packages/auth/src/authClients/supabase.ts

```diff
@@ -40,11 +42,13 @@ export interface AuthClientSupabase extends AuthClient {
    * @param options The user login details.
    * @param options.email The user's email address.
    * @param options.password The user's password.
+   * @param options.phone The user's phone number.
    * @param redirectTo A URL or mobile address to send the user to after they are confirmed.
    */
   signup(options: {
-    email: string
-    password: string
+    email?: string
```

Comment:
Do you think we could change this to be like:

`options: EmailAuth | SmsAuth | RefreshToken | OtherAuth`>

Reply:
Possibly, but I chose to match the Supabase GoTrue client:

https://github.com/supabase/gotrue-js/blob/797ea09b6188291861b5bed4a488cc74b6f95aaf/src/GoTrueClient.ts#L99

```js
  async signUp(
    { email, password, phone }: UserCredentials,
```

The just have one type:

https://github.com/supabase/gotrue-js/blob/797ea09b6188291861b5bed4a488cc74b6f95aaf/src/lib/types.ts#L106

```js
export interface UserCredentials {
  email?: string
  phone?: string
  password?: string
  refreshToken?: string
  // (Optional) The name of the provider.
  provider?: Provider
}
```

I see that it would make sense, but seems more practical to follow the Supabase client lead to keep in sync.

Maybe we should make that suggestion to them, actually.

Reply:
Sure makes sense.